src/lib/Components/Bidding/Components/MarkdownRenderer.tsx

```diff
@@ -1,71 +1,69 @@
 import React from "react"
-import { Text, View } from "react-native"
-import SimpleMarkdown from "simple-markdown"
+import { Text } from "react-native"
 import styled from "styled-components/native"
 
-import { Row } from "../Elements/Grid"
+import { Serif16 } from "../Elements/Typography"
 
 import SwitchBoard from "lib/NativeModules/SwitchBoard"
 
-// Rules for rendering parsed markdown. Currently only handles links and text. Add rules similar to
-// https://github.com/CharlesMangwa/react-native-simple-markdown/blob/next/src/rules.js for new functionalities.
-const rules = {
-  ...SimpleMarkdown.defaultRules,
-  link: {
-    ...SimpleMarkdown.defaultRules.link,
-    react: (node, output, state) => {
-      state.withinText = true
-      let element
-      const openUrl = url => {
-        SwitchBoard.presentModalViewController(element, url)
-      }
-      return (
-        <LinkText key={state.key} onPress={() => openUrl(node.target)} ref={el => (element = el)}>
-          {output(node.content, state)}
-        </LinkText>
-      )
-    },
-  },
-  text: {
-    ...SimpleMarkdown.defaultRules.text,
-    react: (node, _output, state) => {
-      return <Text key={state.key}>{node.content}</Text>
-    },
-  },
-  paragraph: {
-    ...SimpleMarkdown.defaultRules.paragraph,
-    react: (node, output, state) => {
-      return (
-        <Row style={{ flexDirection: "row" }} key={state.key}>
-          {output(node.content, state)}
-        </Row>
-      )
-    },
-  },
-}
-// Markdown parser setup
-const rawBuiltParser = SimpleMarkdown.parserFor(rules)
-const parser = source => {
-  const blockSource = source + "\n\n"
-  return rawBuiltParser(blockSource, { inline: false })
-}
-const reactOutput = SimpleMarkdown.reactFor(SimpleMarkdown.ruleOutput(rules, "react"))
-
-export const toReact = md => {
-  const syntaxTree = parser(md)
-  return reactOutput(syntaxTree)
-}
-
 interface NativeMarkdownProps {
-  md: string
+  children: string
 }
 
+const PARAGRAPH_BREAK = "   "
+
 export class MarkdownRenderer extends React.Component<NativeMarkdownProps> {
   render() {
-    return <View key={0}>{reactOutput(parser(this.props.md))}</View>
+    const message = this.props.children
+    if (!message) {
+      return
+    }
+
+    const paragraphs = message.split(PARAGRAPH_BREAK)
+    const paragraphEls = paragraphs.map((p, i) => {
+      let paragraph = p
+      // regex for [anything](url) though url doesn't need to be strictly validated.
+      const re = new RegExp(/\[([^\]]*)\]\(([-a-zA-Z0-9@:%._\+~#=\:\/]*)\)/g)
```

Comment:
Can we extract this into its own function so we can verify the logic before the other tests are working?

Reply:
With extracting regex I ended up with an almost empty render function which wasn't any better than the previous tests. It is possible to revisit this when taking care of the other tests and make the regex parsing less intertwined. For now I added some tests just for the regex.